bundles/org.openhab.binding.airvisualnode/src/main/java/org/openhab/binding/airvisualnode/internal/discovery/AirVisualNodeDiscoveryService.java

```diff
@@ -105,27 +112,35 @@ private void scan() {
             // Extract the Node serial number from device name
             String nodeSerialNumber = m.group(1);
 
-            // The Node Thing UID is serial number converted to lower case
-            ThingUID thingUID = new ThingUID(AirVisualNodeBindingConstants.THING_TYPE_AVNODE,
-                    nodeSerialNumber.toLowerCase());
-
-            try {
-                // Get the Node address by name
-                NbtAddress nodeNbtAddress = NbtAddress.getByName(serverName);
-                if (nodeNbtAddress == null) {
-                    // The Node address not found by some reason, skip it
-                    continue;
+            if (nodeSerialNumber != null) {
```

Comment:
Consider reducing nesting:
```suggestion
            if (nodeSerialNumber == null) {
                logger.debug("Extracting the Node serial number failed");
                return;
            }
```

Reply:
I think it is just some structure i (wrongly) tought myself. I'll try to remember this for all PR's ill do in the future now ;-)